src/sidebar/components/test/top-bar-test.js

```diff
@@ -1,15 +1,20 @@
 'use strict';
 
 var angular = require('angular');
+var proxyquire = require('proxyquire');
 
 var util = require('../../directive/test/util');
 
 describe('topBar', function () {
   var fakeSettings = {};
+  var fakeIsThirdPartyService = sinon.stub();
```

Comment:
Hmm ... in most other tests we do the initialization of stubs in the `beforeEach` function via assignment. This makes it more obvious that the value is always "fresh" for each test, regardless of type.

Reply:
Yes. Unfortunately `angular.module()` is called in a `before()` that happens _before_ the `beforeEach()` in which the rest of the fakes are initialized. So this one fake needs to be initialized sooner. I've run across this a few times. I think the solution may have to be for the `angular.module()` call or similar (construction of an angular directive, etc) to happen in a `beforeEach()` instead of a `before()`, so that params that get passed into it can be re-assigned per test.